File: go/pubsub/sub.go
// sub.go
package main

import (
	"fmt"

	zmq "github.com/pebbe/zmq4"
)

func main() {
	// Create a new SUB socket
	subSocket, err := zmq.NewSocket(zmq.SUB)
	if err != nil {
		fmt.Println("Couldn't create SUB socket:", err)
		return
	}
	defer subSocket.Close()

	// Connect the SUB socket to the PUB socket
	err = subSocket.Connect("tcp://127.0.0.1:5680")
	if err != nil {
		fmt.Println("Couldn't connect SUB socket:", err)
		return
	}

	// Subscribe to messages with prefix "1"
	subSocket.SetSubscribe("1")

	// Start receiving messages
	for {
		message, err := subSocket.Recv(0)
		if err != nil {
			fmt.Println("Couldn't receive message:", err)
			break
		}
		fmt.Println("Received:", message)
	}
}

File: rust/pubsub/src/sub.rs
extern crate zmq;

fn main() {
    let context = zmq::Context::new();
    let subscriber = context.socket(zmq::SUB).unwrap();

    subscriber.connect("tcp://localhost:5561").unwrap();
    subscriber.set_subscribe("1".as_bytes()).unwrap();

    loop {
        let msg = subscriber.recv_string(0).unwrap().unwrap();
        println!("Received: {}", msg);
    }
}
